File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int product = 1;
        int count =0; 
        int pos;
        for(int i =0;i<nums.size();i++){
            if(nums[i] !=0){
                product = product*nums[i];
            }
            else{
                count++;
                pos = i;
            }
        }
        vector<int> ans(nums.size(),0);
        if(count > 1){
            return ans;
        }
        else if(count == 1){
            ans[pos] = product;
        }
        else{
        for(int i=0;i<nums.size();i++){
                ans[i]=product/nums[i]; 
        }
        }
        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        zeroCount = 0
        multiple = 1
        for i in nums:
            if i == 0:
                zeroCount +=1
            else:
                multiple *=i
        
        newArr = [0]*len(nums)
        if zeroCount==0:
            i = 0
            while(i<len(nums)):
                newArr[i] = multiple//nums[i]
                i +=1
        elif zeroCount==1:
            i = 0
            while(i<len(nums)):
                if nums[i]==0:
                    newArr[i] = multiple
                i +=1
        
        return newArr

